### PR TITLE
chore: improve some error messages

### PR DESCRIPTION
**Pull request by charles-cooper**
### Commit message

```
fix array bounds check and create builtin error messages

- array bounds checks, previously were something like

`clamp lt [mload, 640 <result>]`

- codesize check error message was missing for create builtins
- create failure error message was also missing
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/codegen/ir_node.py [hunk 1]
```diff
@@ -330,6 +330,15 @@ def is_complex_ir(self):
             and self.value.lower() not in do_not_cache
         )
 
+    # set an error message and push down into all children.
+    # useful for overriding an error message generated by a helper
+    # function with a more specific error message.
+    def set_error_msg(self, error_msg: str) -> None:
+        self.error_msg = error_msg
+        for arg in self.args:
+            self.set_error_msg(error_msg)
```
Comment (fubuloubu) [inline vyper/codegen/ir_node.py:339]: What's going on here?

Comment (charles-cooper) [reply vyper/codegen/ir_node.py:339]: it pushes down the error message into all the child IR nodes

File: vyper/codegen/ir_node.py [hunk 2]
```diff
@@ -330,6 +330,15 @@
             and self.value.lower() not in do_not_cache
         )
 
+    # set an error message and push down into all children.
+    # useful for overriding an error message generated by a helper
+    # function with a more specific error message.
+    def set_error_msg(self, error_msg: str) -> None:
+        self.error_msg = error_msg
+        for arg in self.args:
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply vyper/codegen/ir_node.py:338]: Is `arg` supposed to be used?

[the remark this answers is not in the document]
Comment (charles-cooper) [reply vyper/codegen/ir_node.py:338]: yea that was fixed in https://github.com/vyperlang/vyper/pull/3524/commits/3f4a625402bf2ff4aaa791c8e14a9e518f182327

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**